src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -403,6 +410,14 @@ public SubversionSCM(String[] svnUrls, String[] credentialIds, String[] locals)
         this(ModuleLocation.parse(svnUrls, credentialIds, locals, null,null), true, false, null, null, null, null, null);
     }
 
+
+    /**
+     * Convenience constructor, especially during testing.
+     */
+    public SubversionSCM(String[] svnUrls, String[] credentialIds, String[] locals, List<AdditionalCredentials> additionalCredentials) {
```

Comment:
I do not believe new methods are really required if you implement proper null handling in the original constructors. You say these methods are useful for testing, but there is no autotests